### PR TITLE
fix: return non UTF-8 error message

### PR DESCRIPTION
**Pull request by FrankYang0529**
Fixes https://github.com/rust-lang/cargo/issues/11311

### Test Steps

1. Create a new empty Git repository
2. Create a `.gitignore` that is not valid UTF-8, for instance `printf '\xFF\xFE' > .gitignore`
3. `cargo init`

### CODE REVIEW

File: src/cargo/ops/cargo_new.rs [hunk 1]
```diff
@@ -595,9 +595,22 @@ impl IgnoreList {
...
-        let existing_items = existing.lines().collect::<Result<Vec<_>, _>>().unwrap();
+    fn format_existing<T: BufRead>(&self, existing: T, vcs: VersionControl) -> CargoResult<String> {
+        let mut existing_items = Vec::new();
+        for (i, item) in existing.lines().enumerate() {
+            match item {
+                Ok(s) => existing_items.push(s),
+                Err(err) => match err.kind() {
+                    ErrorKind::InvalidData => {
+                        return Err(anyhow!(
+                            "Character at line {} is invalid. Cargo only supports UTF-8.",
```
Comment (epage) [inline src/cargo/ops/cargo_new.rs:606]: Could you add a test for this?

Comment (FrankYang0529) [reply src/cargo/ops/cargo_new.rs:606]: Added it. Thank you!

File: tests/testsuite/new.rs [hunk 2]
```diff
@@ -509,3 +509,25 @@ fn git_default_branch() {
...
     assert_eq!(head.symbolic_target().unwrap(), "refs/heads/hello");
 }
+
+#[cargo_test]
+fn non_utf8_str_in_ignore_file() {
+    let gitignore = paths::home().join(".gitignore");
+    File::create(gitignore).unwrap();
+
+    fs::write(paths::home().join(".gitignore"), &[0xFF, 0xFE]).unwrap();
+    print!("{}", paths::home().display());
```
Comment (hi-rustin) [inline tests/testsuite/new.rs:519]: Could you please explain why we need this print?

File: tests/testsuite/new.rs [hunk 3]
```diff
@@ -509,3 +509,25 @@ fn git_default_branch() {
...
+        .with_status(101)
+        .with_stderr(format!(
+            "\
+error: Failed to create package `home` at `{}`
+
+Caused by:
+  Character at line 0 is invalid. Cargo only supports UTF-8.
+",
+            paths::home().display()
+        ))
```
Comment (hi-rustin) [inline tests/testsuite/new.rs:531]:
```suggestion
        .with_stderr(
            "\
error: Failed to create package `home` at `[..]`

Caused by:
  Character at line 0 is invalid. Cargo only supports UTF-8.
"
)
```

**Review by hi-rustin — COMMENTED** [review]
The rest looks good to me! Thanks!

File: tests/testsuite/new.rs (hunk 2 above)
Comment (FrankYang0529) [reply tests/testsuite/new.rs]: Sorry, this is just for local testing. Removed it. Thank you.

File: tests/testsuite/new.rs (hunk 3 above)
Comment (FrankYang0529) [reply tests/testsuite/new.rs]: Updated it. Thank you.

**Pull request merged**